### PR TITLE
docs: add supported option types for autocomplete

### PR DESCRIPTION
**Pull request by vitorlops**
**Please describe the changes this PR makes and why it should be merged:**

- https://github.com/discord/discord-api-docs/pull/4318

This PR documents the supported option types for autocomplete.

**Status and versioning classification:**

- This PR **only** includes non-code changes, like changes to documentation, README, etc.

### CODE REVIEW

File: packages/discord.js/src/structures/CommandInteraction.js [hunk 1]
```diff
@@ -1,6 +1,7 @@
 'use strict';
 
 const { Collection } = require('@discordjs/collection');
+const { ApplicationCommandOptionType } = require('discord-api-types/v9');
```
Comment (ImRodry) [inline packages/discord.js/src/structures/CommandInteraction.js:4]: why?

Comment (vitorlops) [reply packages/discord.js/src/structures/CommandInteraction.js:4]: I thought this was necessary to use these `@link`s
image

Comment (ImRodry) [reply packages/discord.js/src/structures/CommandInteraction.js:4]: it isn't

Comment (vitorlops) [reply packages/discord.js/src/structures/CommandInteraction.js:4]: Ok, thanks for let me know.
I will fix it as soon as I can.

**Comment by vitorlops** [comment]
Little accident, sorry.

**Review by vladfrangu — APPROVED** [review]

**Review by kyranet — APPROVED** [review]

**Review by iCrawl — APPROVED** [review]

**Pull request merged**